src/RL/utils/geometricflux/heterogeneousgraphconv.jl

```diff
@@ -113,4 +113,93 @@ function (g::HeterogeneousGraphConv{<:AbstractMatrix,<:Any,meanPooling})(fg::Het
+    H1, H2, H3 = fgs.varnf, fgs.connf, fgs.valnf
+    X1, X2, X3 = original_fgs.varnf, original_fgs.connf, original_fgs.valnf
+    contovarN, valtovarN, vartoconN, vartovalN = contovar, valtovar, vartocon, vartoval
+
+    Zygote.ignore() do
+        contovarIdx = repeat(collect(1:size(contovar,1)),1,size(contovar,2)).*contovar
+        valtovarIdx = repeat(collect(1:size(valtovar,1)),1,size(valtovar,2)).*valtovar
+        vartoconIdx = repeat(collect(1:size(vartocon,1)),1,size(vartocon,2)).*vartocon
+        vartovalIdx = repeat(collect(1:size(vartoval,1)),1,size(vartoval,2)).*vartoval
+
+        contovarIdx = collect.(zip(contovarIdx,cat([repeat([i],size(contovar)[1:2]...) for i in 1:size(contovar)[3]]..., dims= 3)))
+        valtovarIdx = collect.(zip(valtovarIdx,cat([repeat([i],size(valtovar)[1:2]...) for i in 1:size(valtovar)[3]]..., dims= 3)))
+        vartoconIdx = collect.(zip(vartoconIdx,cat([repeat([i],size(vartocon)[1:2]...) for i in 1:size(vartocon)[3]]..., dims= 3)))
+        vartovalIdx = collect.(zip(vartovalIdx,cat([repeat([i],size(vartoval)[1:2]...) for i in 1:size(vartoval)[3]]..., dims= 3)))
+
+        filteredcolcontovar = mapslices( x -> map(z-> filter(y -> y[1]!=0, z), eachcol(x)), contovarIdx, dims=[1,2])
+        filteredcolvaltovar = mapslices( x -> map(z-> filter(y -> y[1]!=0, z), eachcol(x)), valtovarIdx, dims=[1,2])
+        filteredcolvartocon = mapslices( x -> map(z-> filter(y -> y[1]!=0, z), eachcol(x)), vartoconIdx, dims=[1,2])
+        filteredcolvartoval = mapslices( x -> map(z-> filter(y -> y[1]!=0, z), eachcol(x)), vartovalIdx, dims=[1,2])
+
+        filteredembcontovar = mapslices(x->map(y-> !isempty(y) ? reshape(Base.maximum(mapreduce(z->H2[:,z...], hcat, y), dims =2),:) : zeros(Float32,size(H2,1)),  x), filteredcolcontovar, dims = [1,3])
+        filteredembvaltovar = mapslices(x->map(y-> !isempty(y) ? reshape(Base.maximum(mapreduce(z->H3[:,z...], hcat, y), dims =2), :) : zeros(Float32,size(H3,1)),  x), filteredcolvaltovar, dims = [1,3])
+        filteredembvartocon = mapslices(x->map(y-> !isempty(y) ? reshape(Base.maximum(mapreduce(z->H1[:,z...], hcat, y), dims =2), :) : zeros(Float32,size(H1,1)),  x), filteredcolvartocon, dims = [1,3])
+        filteredembvartoval  = mapslices(x->map(y-> !isempty(y) ? reshape(Base.maximum(mapreduce(z->H1[:,z...], hcat, y), dims =2),:) : zeros(Float32,size(H1,1)),  x), filteredcolvartoval , dims = [1,3])
+        filteredembcontovar = reshape(reduce(hcat ,filteredembcontovar), :,size(H1)[2:3]...)
+        filteredembvaltovar = reshape(reduce(hcat ,filteredembvaltovar), :,size(H1)[2:3]...)
+        filteredembvartocon = reshape(reduce(hcat ,filteredembvartocon), :,size(H2)[2:3]...)
+        filteredembvartoval = reshape(reduce(hcat ,filteredembvartoval), :,size(H3)[2:3]...)
+
+    return BatchedHeterogeneousFeaturedGraph{Float32}(
+        contovar,
+        valtovar,
+        g.σ.(g.weightsvar ⊠ vcat(X1, H1, filteredembcontovar, filteredembvaltovar) .+ g.biasvar),
```

Comment:
Operations here should be defined outside Zygote.ignore() to allow for backpropagation.